### PR TITLE
adding create corpus and train model files

### PR DESCRIPTION
**Pull request by Iddoyadlin**

### CODE REVIEW

File: research/create_corpus.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            data = r.read(chunk_size)
+            if not data:
+                break
+            out.write(data)
+    r.release_conn()
+
+
+if __name__ == "__main__":
+    download()
+    inp = datapath(r'C:\Users\Iddo\Documents\semantle-he\research\hewiki-latest-pages-articles.xml.bz2')
```
Comment (ishefi) [inline research/create_corpus.py:29]: :)

File: research/create_corpus.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    outp = "wiki.he.text"
+    print("Starting to create wiki corpus")
+
+    space = " "
+    wiki = WikiCorpus(inp, dictionary={})
+    with open(outp, 'wb') as output:
+        for i, text in enumerate(wiki.get_texts(), start=1):
+            article = " ".join(text)
+            output.write("{}\n".format(article).encode('utf-8'))
+            if (i % 1000 == 0):
```
Comment (ishefi) [inline research/create_corpus.py:39]: you said there are tons of articles - won't printing this every 1000 articles spam the screen?

(maybe add a `--quite` option as well?)

File: research/train_model.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import multiprocessing
+
+if __name__=="__main__":
+    model_name = "model.mdl"
+    model_input = "wiki.he.text"
```
Comment (ishefi) [inline research/train_model.py:5]: maybe allow adding these as arguments? 

same question for the previous script

(or at least save them in `consts.py`)

File: research/train_model.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+    from gensim.models import Word2Vec
+    from gensim.models.word2vec import LineSentence
+    model = Word2Vec(LineSentence(model_input), sg=1, vector_size=100, window=5, min_count=20, workers=multiprocessing.cpu_count())
+    model.init_sims(replace=True)
+    model.save(model_name)
```
Comment (ishefi) [inline research/train_model.py:11]: no new line a the end of the file!

File: research/train_model.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import multiprocessing
+
+if __name__=="__main__":
+    model_name = "model.mdl"
+    model_input = "wiki.he.text"
+
+    from gensim.models import Word2Vec
+    from gensim.models.word2vec import LineSentence
```
Comment (ishefi) [inline research/train_model.py:8]: why are the imports here?

File: research/train_model.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+    model = Word2Vec(LineSentence(model_input), sg=1, vector_size=100, window=5, min_count=20, workers=multiprocessing.cpu_count())
+    model.init_sims(replace=True)
```
Comment (ishefi) [inline research/train_model.py:10]: I think this deprecated

File: research/create_corpus.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    r.release_conn()
+
+
+if __name__ == "__main__":
+    download()
+    inp = datapath(r'C:\Users\Iddo\Documents\semantle-he\research\hewiki-latest-pages-articles.xml.bz2')
+    outp = "wiki.he.text"
+    print("Starting to create wiki corpus")
+
+    space = " "
```
Comment (ishefi) [inline research/create_corpus.py:33]: this is true, but I think you don't use it anywhere

File: research/create_corpus.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    path = 'hewiki-latest-pages-articles.xml.bz2'
+
+    if os.path.exists(path):
+        print("skipping download")
+        return
+    chunk_size = 1024
+    http = urllib3.PoolManager(cert_reqs='CERT_NONE')
+    r = http.request('GET', url, preload_content=False)
+    with open(path, 'wb') as out:
+        while True:
```
Comment (ishefi) [inline research/create_corpus.py:19]: what about:
```python
data = r.read(chunk_size)
while data:
    out.write(data)
    data = r.read(chunk_size)    
```

**Review by ishefi — CHANGES_REQUESTED** [review]

File: research/train_model.py (hunk 3 above)
Comment (Iddoyadlin) [reply research/train_model.py]: f097ba782e8ed291798ff98478a061bfedf71e92

File: research/train_model.py (hunk 4 above)
Comment (Iddoyadlin) [reply research/train_model.py]: fixed :)

File: research/train_model.py (hunk 5 above)
Comment (Iddoyadlin) [reply research/train_model.py]: fixed :)

File: research/create_corpus.py (hunk 7 above)
Comment (Iddoyadlin) [reply research/create_corpus.py]: fixed :)

File: research/create_corpus.py (hunk 8 above)
Comment (Iddoyadlin) [reply research/create_corpus.py]: 3f358a37948c7dced53635f86dc8ae624aa6eba8

File: research/train_model.py (hunk 6 above)
Comment (Iddoyadlin) [reply research/train_model.py:11]: its kind of strange... it seems deprecated yet it does normalize all distances... whatever you think

**Comment by Iddoyadlin** [comment]
I think the output of the research folder (after all training, postprocessing etc.) should be just the words and their vectors. these can be saved in the format you did for loading to mongo (what I currently implemented), or we can use Word2vec.save_word2vec_format (which is some standard for saving word2vecs).

I'd like to be able to train using a library different than gensim

File: research/create_corpus.py (hunk 1 above)
Comment (Iddoyadlin) [reply research/create_corpus.py]: yeah my bad... changed

File: research/create_corpus.py (hunk 2 above)
Comment (Iddoyadlin) [reply research/create_corpus.py]: 7b3188c0830f8e050cd809a4c360d3a72057bf93

**Review by ishefi — APPROVED** [review]

**Pull request merged**